components/net/http_loader.rs

```diff
@@ -2320,18 +2320,30 @@ fn serialize_request_origin(request: &Request) -> headers::Origin {
     }
 
     // Step 3. Return request’s origin, serialized.
+    serialize_origin(origin)
+}
+
+/// Step 3 of <https://fetch.spec.whatwg.org/#serializing-a-request-origin>.
+pub fn serialize_origin(origin: &ImmutableOrigin) -> headers::Origin {
     match origin {
         ImmutableOrigin::Opaque(_) => headers::Origin::NULL,
         ImmutableOrigin::Tuple(scheme, host, port) => {
+            // Note: This must be kept in sync with `Origin::ascii_serialization()``, which does not
```

Comment:
Is this change still in conformance with the spec, or is it expected to be removed once we update the code that uses `ascii_serialization()` (or fixed in rust-url)?

Reply:
The specifications seem to differ on this. For instance, the original RFC 6454 specification says to omit the port if it's the default port (https://www.rfc-editor.org/rfc/rfc6454#section-6.2) and this is what Chromium does (https://chromium.googlesource.com/chromium/src/+/lkgr/url/scheme_host_port.cc#296). Older versions of the HTML specification had language like this as well (https://www.w3.org/TR/2011/WD-html5-20110525/origin-0.html), but the current one does not (https://html.spec.whatwg.org/#ascii-serialisation-of-an-origin).

It looks like what happens is that the data model of the URL changed such that internally the origin's port can be null when it's the default one (https://github.com/whatwg/html/pull/870). The origin in `rust-url` doesn't have support for this yet (https://github.com/servo/rust-url/issues/821). Once that is fixed this should work properly without an extra check.

For us, the most important thing is that the serialization is consistent, I think.

Reply:
That makes sense! Thanks for the additional context!